### ISSUE TITLE
Favicon download is broken

### ISSUE THREAD

**Issue by saschabrockel** [keeweb/keeweb#2108]
**Describe the bug**

**To Reproduce**
1. Go to your password file and log in
2. Click on an entry with an website attached and maybe check if the website contains a favicon.ico in its domain like www.themoviedb.org
3. Click the favicon download button

**Expected behavior**
It should download the favicon but returns an error.

**Screenshots**

![image](https://github.com/keeweb/keeweb/assets/33376463/5476f759-259d-42cd-ae89-245ee5610d41)

**Environment**
Please copy all version information from Settings/Help

**Kdbx File**
Does it happen on Demo or New database?  
Everywhere.

**Logs**
Please open dev tools in your browser and attach output log from Console tab. If you are using a desktop app, devtools can be opened from Settings/General/Advanced.
```
services.keeweb.info/favicon/www.themoviedb.org:1     GET https://services.keeweb.info/favicon/www.themoviedb.org net::ERR_FAILED 504
Image (async)
downloadIcon @ keeweb?config=config?file=%2FPasswords.kdbx:1
eventListener @ keeweb?config=config?file=%2FPasswords.kdbx:1
e @ keeweb?config=config?file=%2FPasswords.kdbx:1
keeweb?config=config?file=%2FPasswords.kdbx:1 2023-06-23T21:04:32.400Z [icon-select-view] Favicon download error: https://services.keeweb.info/favicon/www.themoviedb.org Event {isTrusted: true, type: 'error', target: img, currentTarget: img, eventPhase: 2, …}
r.error @ keeweb?config=config?file=%2FPasswords.kdbx:1
t.onerror @ keeweb?config=config?file=%2FPasswords.kdbx:1
error (async)
downloadIcon @ keeweb?config=config?file=%2FPasswords.kdbx:1
eventListener @ keeweb?config=config?file=%2FPasswords.kdbx:1
e @ keeweb?config=config?file=%2FPasswords.kdbx:1
```

**Comment by clicky6** [comment]
+1
This issue is a major one and it doesn't look too hard to fix, Anyone up for fixing it?

**Comment by Aetherinox** [comment]
KeeWeb originally hosted the favicon service which allowed for icons to be grabbed and downloaded to your vault. Unfortunately, some users abused this system.

I'm currently looking into 3rd party services which will allow us to utilize their API to do the same thing.

**Comment by clicky6** [comment]
> KeeWeb originally hosted the favicon service which allowed for icons to be grabbed and downloaded to your vault. Unfortunately, some users abused this system.
> 
> I'm currently looking into 3rd party services which will allow us to utilize their API to do the same thing. Once I find the right fit, I'll integrate it into the next version.

Can do it like KeepassXC does...

**Comment by Aetherinox** [comment]
As of now, I have our service back up. Except it's now hosted on a Cloudflare Worker.

The downside is that Cloudflare does not support polyfill / http / https. So it made the original library antelle developed not work, and I had to write one specifically for Cloudflare.

https://keeweb-favicon.aetherinox.workers.dev/
https://keeweb-favicon.aetherinox.workers.dev/?url=https://keeweb.info

I've debated DDG as a backup, but I need to review a few things first.
